clients/js/algorand.ts

```diff
@@ -0,0 +1,164 @@
+import {
+  CONTRACTS,
+  getIsTransferCompletedAlgorand,
+  redeemOnAlgorand,
+} from "@certusone/wormhole-sdk";
+import { NETWORKS } from "./networks";
+import { impossible, Payload } from "./vaa";
+import {
+  Account,
+  Algodv2,
+  assignGroupID,
+  mnemonicToSecretKey,
+  waitForConfirmation,
+} from "algosdk";
+import { TransactionSignerPair } from "@certusone/wormhole-sdk/lib/cjs/algorand";
+
+async function signSendAndConfirmAlgorand(
+  algodClient: Algodv2,
+  txs: TransactionSignerPair[],
+  wallet: Account
+) {
+  assignGroupID(txs.map((tx) => tx.tx));
+  const signedTxns: Uint8Array[] = [];
+  for (const tx of txs) {
+    if (tx.signer) {
+      signedTxns.push(await tx.signer.signTxn(tx.tx));
+    } else {
+      signedTxns.push(tx.tx.signTxn(wallet.sk));
+    }
+  }
+  await algodClient.sendRawTransaction(signedTxns).do();
+  const result = await waitForConfirmation(
+    algodClient,
+    txs[txs.length - 1].tx.txID(),
+    1
+  );
+  return result;
+}
+
+export async function execute_algorand(
+  payload: Payload,
+  vaa: Uint8Array,
+  environment: "MAINNET" | "TESTNET" | "DEVNET"
+) {
+  const ALGORAND_HOST =
+    environment === "MAINNET"
+      ? {
+          algodToken: "",
+          algodServer: "https://mainnet-api.algonode.cloud",
+          algodPort: "",
+        }
+      : environment === "TESTNET"
+      ? {
+          algodToken: "",
+          algodServer: "https://testnet-api.algonode.cloud",
+          algodPort: "",
+        }
+      : {
+          algodToken:
+            "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+          algodServer: "http://localhost",
+          algodPort: "4001",
+        };
+  const chainName = "algorand";
+  let n = NETWORKS[environment][chainName];
+  if (!n.key) {
+    throw Error("No " + environment + " rpc defined for Algorand");
+  }
+  let contracts = CONTRACTS[environment][chainName];
+
+  let target_contract: string;
+
+  switch (payload.module) {
+    case "Core":
+      target_contract = contracts.core;
+      // sigh...
+      switch (payload.type) {
+        case "GuardianSetUpgrade":
+          console.log("Submitting new guardian set");
+          break;
+        case "ContractUpgrade":
+          console.log("Upgrading core contract");
+          break;
+        default:
+          impossible(payload);
+      }
+      break;
+    case "NFTBridge":
+      if (contracts.nft_bridge === undefined) {
```

Comment:
```suggestion
      if (typeof contracts.nft_bridge === "undefined") {
```

I BELIEVE that using typeof is the idiomatic javascript-y way of doing this, but @kev1n-peters or @evan-gray can tell me that's wrong.

Reply:
Leaving as is per @evan-gray